### PR TITLE
Adding JSONReporter to the SDK for CI analytics

### PR DESCRIPTION
**Pull request by guinaut**
## Description
Adds a JSONReporter class that outputs evaluation results to a reports folder when called through the CLI.

Example usage:
```
  eval_1 = model_under_test.run_test( ... )
  eval_2 = model_under_test.run_test( ... )
  reporter = JSONReporter([eval_1, eval_2])
  reporter.log()
```

### CODE REVIEW

File: src/okareo/reporter.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import json
+import os
+
+from okareo_api_client.models import TestRunItem
+
+
+class JSONReporter:
+    """
+    A simmple class to define a json reporter for output of results.
```
Comment (mdelrosa) [inline src/okareo/reporter.py:9]: typo: simple

File: src/okareo/reporter.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    """
+
+    def __init__(self, eval_runs: list[TestRunItem]):
+        self.eval_runs = eval_runs
+
+    def log(self) -> None:
+        """
+        Logs the results of the evaluation runs into a file based on OKAREO_REPORT_DIR.
+        """
+        okareo_report_dir = os.environ["OKAREO_REPORT_DIR"]
```
Comment (mdelrosa) [inline src/okareo/reporter.py:20]: if this env var isn't defined, does this just write the report to the current directory?

**Review by mdelrosa — APPROVED** [review]
Approved with one nit and one question.

Would like to see tests that are at parity with [reporters.test.ts](https://github.com/okareo-ai/okareo-typescript-sdk/blob/main/tests/reporters.test.ts), but should be fine to merge and follow-up with the tests.

File: src/okareo/reporter.py (hunk 2 above)
Comment (guinaut) [reply src/okareo/reporter.py:20]: This env_var is from the CLI.  When it is not present, this outputs to the console.

**Pull request merged**